src/App.tsx

```diff
@@ -15,37 +17,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SetByReverse = true;
+
+enum SortedFields {
+  SetById,
+  SetByAlphabet,
+  SetByLength,
+  SetByDanger,
+}
+
+function getSortedGoods(
+  goods: string[],
+  { sortField, reversed }: { sortField: SortedFields; reversed: boolean },
+) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortField) {
+      case SortedFields.SetById:
+      case SortedFields.SetByAlphabet:
+        return good1.localeCompare(good2);
+      case SortedFields.SetByLength:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setsortField] = useState(SortedFields.SetByDanger);
```

Comment:
The function name 'setsortField' does not follow the naming convention for functions. Function names should start with a verb and use camelCase. Consider renaming it to 'setSortField' to improve readability.